client/include/frontier_client/frontier.h

```diff
@@ -78,6 +77,31 @@ void frontier_free(void *ptr);
 // GZip and base64URL encode
 int fn_gzip_str2urlenc(const char *str,int size,char **out);
 
+// NOTE: The contents of this struct may not change or binary compatibility
+//   will break.
+struct s_FrontierStatisticsNum
+ {
+  unsigned int min;
+  unsigned int max;
+  unsigned int avg;
+ };
+typedef struct s_FrontierStatisticsNum FrontierStatisticsNum;
+
+// NOTE: for binary compatibility, variables should never be removed
+//   from this structure, and new ones should only be added to the end.
```

Comment:
How important is binary compatibility?

If it's a concern, it might be better to use the [PIMPL](https://en.cppreference.com/w/cpp/language/pimpl) pattern to hide the size of the struct.  It already seems from below that you intend it to only be accessed from a standalone function.

Reply:
I think binary compatibility is very important, so the library can be upgraded without requiring application code that's using it to have to be recompiled.  The code is in a shared library, and they're supposed to change the major number if they break backward compatibility (that is, go from libfrontier_client.so.2 to libfrontier_client.so.3).

If I understand the PIMPL pattern correctly, it requires only accessing the data through functions that are supplied by the library code.  That's much easier to do in C++ than C, and it seems unnecessarily complex in C.  Yes I intend for it only to be read out by frontier_statistics_get(), but once that's done the individual items are accessed directly by the application.